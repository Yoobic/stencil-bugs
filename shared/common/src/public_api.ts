export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';