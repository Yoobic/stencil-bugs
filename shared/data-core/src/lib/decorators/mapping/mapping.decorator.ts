import { Models } from '../../services/models/models.service';

export function Mapping(className: string, config: { order: number }) {

    return function decoratorFactory(target: Object, decoratedPropertyName: string): void {
        if (Models) {
            Models.addMappingField(className, decoratedPropertyName, config.order);
        }
    };
}
