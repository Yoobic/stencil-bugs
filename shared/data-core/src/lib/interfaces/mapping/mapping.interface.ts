import { Model } from '../../decorators/model/model.decorator';
import { Editable } from '../../decorators/editable/editable.decorator';
import { IIMapping, FormFieldType } from '@shared/interfaces';
import { ADMIN_FILES_TYPE, FORMCREATOR_FILES_TYPE } from '../constants/constants.interface';

let conditions = {
    isAdmin: 'context == "admin"',
    isFormCreator: 'context == "formCreator"',
    isNotTranslation: 'context!="translation"'
};

@Model({ className: 'IMapping' })
export class IMapping extends IIMapping {
    @Editable('IMapping', {
        required: true,
        autoselect: true,
        type: FormFieldType.autocomplete,
        conditionalValues: [{ condition: conditions.isAdmin, ADMIN_FILES_TYPE }, { condition: conditions.isFormCreator, values: FORMCREATOR_FILES_TYPE }],
        condition: conditions.isNotTranslation,
        defaultValues: ADMIN_FILES_TYPE,
        translate: true
    })
    type: string;

    @Editable('IMapping', { required: true, type: FormFieldType.documentuploader, filterable: false, title: 'DOCUMENT', extensions: ['csv', 'application', 'xls', 'xlsx'] }) //'xls', 'xlsx', //'xls', 'xlsx', , 'application'
    document: any;

}