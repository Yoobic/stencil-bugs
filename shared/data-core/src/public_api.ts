export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
