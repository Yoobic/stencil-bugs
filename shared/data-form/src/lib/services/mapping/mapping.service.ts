import { Injectable, ChangeDetectorRef } from '@angular/core';

import { Translate } from '@shared/translate';
import { Location, IMapping, Xlsx, Condition, Translation, SIMPLE_FIELD_TYPES, Config, Broker, Requestor, Googlemaps, Slide } from '@shared/data-core';
import { IFormField } from '@shared/interfaces';

import { FormDynamicBuilder } from '../form-dynamic-builder/form-dynamic-builder.service';
import { Conditions } from '../conditions/conditions.service';

import { Observable, of, forkJoin } from 'rxjs';
import { mergeMap, map } from 'rxjs/operators';
import { forEach, find, findIndex, indexOf, filter, trimEnd, trimStart } from 'lodash-es';

@Injectable()
export class Mapping {
    constructor(private rq: Requestor, private googlemaps: Googlemaps, private broker: Broker, private conditionsService: Conditions, private config: Config, private translate: Translate) { }

    getTotal(mapping: IMapping, xlsx: Xlsx) {
        if (mapping.document) {
            return xlsx.read(mapping.document).then((data: any) => {
                return data.length - 1;
            });
        } else {
            return Promise.resolve(0);
        }
    }

    convertMissionForExcelExport(slides: Array<any>, conditions: Array<any>) {
        let conditionsData = this.convertConditonsForExcelExport(conditions);
        let slidesData = this.convertSlidesForExcelExport(slides);
        return [slidesData, conditionsData];
    }

    convertSlidesForExcelExport(slides: Array<any>) {
        let columns = [{ name: 'id' }, { name: 'pageTitle' }, { name: 'pageDescription' }, { name: 'title' },
        { name: 'description' }, { name: 'type' }, { name: 'allowComments' }, { name: 'required' },
        { name: 'values' }, { name: 'allowLibrary' }, { name: 'min' }, { name: 'max' },
        { name: 'step' }, { name: 'minDate' }, { name: 'maxDate' }, { name: 'hideMobile' },
        { name: 'allowHistory' }, { name: 'showOnLocation' }, { name: 'allowAnnotate' }, { name: 'showUsers' },
        { name: 'showService' }, { name: 'allowOther' }, { name: 'radius' }, { name: 'multiple' },
        { name: 'clearable' }, { name: 'duration' }, { name: 'anwser' }, { name: 'explanation' },
        { name: 'conditions' }, { name: 'allowTime' }, { name: 'image' }
        ];
        let data = [];
        //let counter = 1;
        forEach(slides, function (slide) {
            forEach(slide.items, function (field: any) {
                data.push({
                    id: field.name,
                    pageTitle: slide.title,
                    pageDescription: slide.description,
                    title: field.title,
                    description: field.type === 'information' ? field.value : field.description,
                    type: field.type,
                    allowComments: field.allowComments,
                    required: field.required,
                    values: field.values,
                    allowLibrary: field.allowLibrary,
                    min: field.min,
                    max: field.max,
                    step: field.step,
                    minDate: field.minDate,
                    maxDate: field.maxDate,
                    hideMobile: field.hideMobile,
                    allowHistory: field.allowHistory,
                    showOnLocation: field.showOnLocation,
                    allowAnnotate: field.allowAnnotate,
                    showUsers: field.showUsers,
                    showService: field.showService,
                    allowOther: field.allowOther,
                    radius: field.radius,
                    multiple: field.multiple,
                    clearable: field.clearable,
                    duration: field.duration,
                    anwser: field.anwser,
                    explanation: field.explanation,
                    conditions: field.condition ? field.condition.map(c => c.title) : '',
                    allowTime: field.allowTime,
                    image: field.image
                });
            });
        });
        return { columns: columns, data: data, title: 'Slides' };
    }

    convertConditonsForExcelExport(conditions: Array<any>) {
        let columns = [
            { name: 'title' },
            { name: 'type' },
            { name: 'field' },
            { name: 'operator' },
            { name: 'values' },
            { name: 'id' }
        ];
        let data = [];
        forEach(conditions, function (condition) {
            data.push({
                title: condition.title,
                type: condition.type,
                field: condition.field ? condition.field.name : '',
                operator: condition.operator,
                values: condition.values || condition.value || condition.tags || condition.group,
                id: condition._id
            });
        });
        return { columns: columns, data: data, title: 'Conditions' };
    }

    upload(mapping: IMapping) {
        return this.rq.postFile(this.config.mappingUrl + mapping.type, mapping.document);
    }

    convertTranslations(data: Array<Array<any>>) {
        data.shift();
        let translations = [];
        forEach(data, (row, i) => {
            if (row[1]) {
                let translation: Translation = {
                    language: row[1],
                    key: row[2],
                    value: row[3],
                    en: row[4],
                    us: row[5],
                    fr: row[6],
                    es: row[7],
                    pl: row[8],
                    nl: row[9],
                    de: row[10],
                    it: row[11],
                    ru: row[12],
                    zhs: row[13],
                    zht: row[14],
                    pt: row[15],
                    kr: row[16],
                    ja: row[17],
                    ua: row[18],
                    he: row[19],
                    ar: row[20],
                    cz: row[21],
                    th: row[22],
                    tr: row[23],
                    bg: row[24],
                    el: row[25],
                    sl: row[26],
                    sk: row[27],
                    isReject: this.isTrue(row[28]),
                    group: this._stringToArray(row[29]),
                    isPhotoAnnotation: this.isTrue(row[30])
                };
                if (row[0]) {
                    translation._id = row[0];
                }
                translations.push(translation);
            }
        });
        return translations;
    }

    convertLocations(data: Array<Array<any>>, progress: { count: number; total: number }, cd: ChangeDetectorRef) { //: Observable<Array<Location>>
        data.shift();
        progress.total = data.length;
        let typeNames: Array<string> = [];
        data.forEach(row => {
            if (typeNames.indexOf(row[2]) < 0) {
                typeNames.push(row[2]);
            }
        });

        return this.broker.getAll('locationtypes', null, null, null, [[{ field: 'name', operator: { _id: 'inq' }, value: typeNames }]]).pipe(
            map(res => res.data),
            mergeMap((types: Array<any>) => {
                let obs: Array<Observable<Location>> = [];
                data.forEach((row, index) => {
                    if (row[0] && row[1]) {
                        let location = new Location();
                        location.title = row[0];
                        location.address = row[1];
                        location.placesearch = row[4] ? row[4] : location.title; // + ' ' + location.address;
                        location.contactname = row[5];
                        location.contactemail = row[6];
                        location.contactphone = row[7];
                        location.info = row[8];
                        location.tags = this._stringToArray(row[9]);
                        location.notificationemail = row[10];
                        location.clientid = row[11];
                        if (row[12]) {
                            location._id = row[12];
                        } else {
                            location._id = 'mapping_' + index;
                        }
                        if (row[13]) {
                            location.vip = this.isTrue(row[13]);
                        }
                        location.type = types.find(t => t.name === row[2]);
                        if (row[14]) {
                            location.missiondescriptionsRef = this._stringToArray(row[14]);
                        }
                        if (!row[3]) {
                            let o = this.googlemaps.resolveAddressLocation(location.address, this.translate.getCurrentLanguage(), location.placesearch).pipe(
                                map(value => {
                                    if (value && value.coords) {
                                        location.status = value.source;
                                        location._geoloc = [value.coords.lng, value.coords.lat];
                                    } else {
                                        location.status = 'error';
                                    }
                                    progress.count += 1;
                                    cd.markForCheck();
                                    return location;
                                }));
                            obs.push(o);
                        } else {
                            location._geoloc = JSON.parse(row[3]);
                            location.status = 'file';
                            progress.count += 1;
                            cd.markForCheck();
                            obs.push(of(location));
                        }
                    }
                });
                return forkJoin(obs);
            }));
    }

    convertFormAndConditions(data: Array<Array<Array<any>>>, formDynamicBuilder: FormDynamicBuilder): { slides: Array<Slide>, conditions: Array<Condition>, errors: Array<any> } {
        let convertError = [];
        let slides = <Array<Slide>>[];
        let conditions = [];
        let fieldsDef = this.removeEmptyRows(data[0]);
        let conditionsDef = data.length >= 2 ? this.removeEmptyRows(data[1]) : [];
        fieldsDef.shift();
        conditionsDef.shift();
        let currentSlide = <Slide>{};
        let fields = new Map<string, IFormField>();
        let conditionedFields = new Map<string, { conditions: Array<any>, slideId: number }>();
        fieldsDef.forEach(row => {
            if (row[1] && currentSlide.title !== row[1]) {
                currentSlide = <Slide>{ title: row[1], description: row[2], items: [] };
                slides.push(currentSlide);
            }
            if (currentSlide) {
                if (row[5]) {
                    let f = this.createField(row, formDynamicBuilder);
                    if (row[28]) {
                        let conditionsTitles = this._stringToArray(row[28]);
                        if (conditionsTitles.length > 0) {
                            let cara = {
                                conditions: conditionsTitles,
                                slideId: slides.length - 1
                            };
                            conditionedFields.set(f.name, cara);
                        }
                    }
                    currentSlide.items.push(f);
                    fields.set(row[0], f);
                }
            }
        });
        conditionsDef.forEach(condition => {
            let c = this.createCondition(condition, fields);
            if (this.conditionsService.isValid(c)) {
                conditions.push(c);
            } else {
                convertError.push('CONVERTCONDITIONERROR : ' + condition[0]);
            }
        });
        conditionedFields.forEach((value, key) => {
            let i = findIndex(slides[value.slideId].items, (f: IFormField) => f.name === key);
            if (i >= 0) {
                slides[value.slideId].items[i].condition = [];
                value.conditions.forEach(title => {
                    let cond = find(conditions, (c) => c.title === title);
                    if (cond) {
                        (<Array<Condition>>slides[value.slideId].items[i].condition).push(cond);
                    }
                });
            }
        });
        return { slides: slides, conditions: conditions, errors: convertError };
    }

    private createField(row: Array<any>, formDynamicBuilder: FormDynamicBuilder): IFormField {
        let field = <IFormField>{
            name: row[0] || formDynamicBuilder.generateFieldName(),
            type: <string>row[5].toLowerCase(),
            title: row[3],
            description: row[4],
            allowComments: this.isTrue(row[6]),
            required: this.isTrue(row[7]),
            values: this._stringToArray(row[8]),
            allowLibrary: this.isTrue(row[9]),
            min: row[10],
            max: row[11],
            step: row[12],
            minDate: row[13],
            maxDate: row[14],
            hideMobile: this.isTrue(row[15]),
            allowHistory: this.isTrue(row[16]),
            showOnLocation: this.isTrue(row[17]),
            allowAnnotate: this.isTrue(row[18]),
            showUsers: this.isTrue(row[19]),
            showService: this.isTrue(row[20]),
            allowOther: this.isTrue(row[21]),
            radius: row[22],
            multiple: this.isTrue(row[23]),
            clearable: this.isTrue(row[24]),
            duration: row[25],
            anwser: this._stringToArray(row[26]),
            explanation: row[27],
            allowTime: this.isTrue(row[29])
        };
        if (row[30] && field.type === 'image') {
            field.image = { _downloadURL: row[30] };
        }
        if (field.type === 'information') {
            field.value = field.description;
            delete field.description;
        }
        return field;
    }

    private createCondition(row: Array<any>, fields: Map<string, IFormField>) {
        let c = <Condition>{
            title: row[0]
        };
        c.operator = this.conditionsService.convertOperator(row[3]);
        c.type = this.conditionsService.convertType(row[1]);
        if (row[5]) {
            c._id = row[5];
        }
        if (c.type === 'field') {
            c.field = this.conditionsService.convertField(fields.get(row[2]));
            if (!c.field) {
                return;
            }
            if (indexOf(SIMPLE_FIELD_TYPES, c.field.type) >= 0) {
                c.value = row[4];
            } else {
                c.values = this._stringToArray(row[4]);
            }
        } else if (c.type === 'tags') {
            c.tags = this._stringToArray(row[4]);
        } else if (c.type === 'groups') {
            c.group = this._stringToArray(row[4]);
        } else {
            return;
        }
        return c;
    }

    private removeEmptyRows(array: Array<Array<string>>): Array<Array<string>> {
        return filter(array, (row) => {
            return filter(row, v => v && !!v.replace(/\s/g, '')).length > 0;
        });
    }

    private _stringToArray = function (str) {
        let array = str ? trimEnd(trimStart(str, '['), ']').split(',') : [];
        for (let i = 0; i < array.length; i++) {
            if (/^\s/.test(array[i])) {
                array[i] = array[i].substring(1);
            }
        }
        return array;
    };

    private isTrue(str) {
        return str === 'TRUE' || str === 'VRAI' || str === 'true' || str === 'vrai';
    }
}
