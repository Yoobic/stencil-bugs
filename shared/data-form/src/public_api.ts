
export { DataFormModule } from './lib/data-form.module';

export * from './lib/interfaces/custom-form-field/custom-form-field.interface';
export * from './lib/interfaces/form-field/form-field.class';
export * from './lib/interfaces/gridstate/gridstate.interface';
export * from './lib/interfaces/cache/cache.interface';
export * from './lib/interfaces/custom-model/custom-model.interface';

export * from './lib/services/cell-renderer/cell-renderer.service';
export * from './lib/services/chart/chart.service';
export * from './lib/services/form-dynamic-builder/form-dynamic-builder.service';
export * from './lib/services/conditions/conditions.service';
export * from './lib/services/mapping/mapping.service';
export * from './lib/services/custom-models/custom-models.service';

export * from './lib/validators/form-filter-field-validator/form-filter-field-validator';
export * from './lib/validators/form-input-base-email-validator/form-input-base-email-validator';
export * from './lib/validators/form-input-base-number-validator/form-input-base-number-validator';
export * from './lib/validators/form-input-between-validator/form-input-between-validator';
export * from './lib/validators/form-input-checklist-validator/form-input-checklist-validator';
export * from './lib/validators/form-input-missionfield-validator/form-input-missionfield-validator';
export * from './lib/validators/form-input-multiphotos-validator/form-input-multiphotos-validator';
export * from './lib/validators/form-input-photo-validator/form-input-photo-validator';
export * from './lib/validators/form-input-select-validator/form-input-select-validator';
export * from './lib/validators/password-are-not-matching-validator/password-are-not-matching-validator';

export * from './lib/renderers/address/address-renderer.component';
export * from './lib/renderers/boolean/boolean-renderer.component';
export * from './lib/renderers/button/button-renderer.component';
export * from './lib/renderers/checklist/checklist-renderer.component';
export * from './lib/renderers/conformity-progress/conformity-progress-renderer.component';
export * from './lib/renderers/conformity-relative-progress/conformity-relative-progress-renderer.component';
export * from './lib/renderers/custom-model/custom-model-renderer.component';
export * from './lib/renderers/datetime/datetime-renderer.component';
export * from './lib/renderers/default/default-renderer.component';
export * from './lib/renderers/delta/delta-renderer.component';
export * from './lib/renderers/distance/distance-renderer.component';
export * from './lib/renderers/duration/duration-renderer.component';
export * from './lib/renderers/location/location-renderer.component';
export * from './lib/renderers/mission-status/mission-status-renderer.component';
export * from './lib/renderers/mission-validation/mission-validation-renderer.component';
export * from './lib/renderers/number/number-renderer.component';
export * from './lib/renderers/multiphotos/multiphotos-renderer.component';
export * from './lib/renderers/percentage/percentage-renderer.component';
export * from './lib/renderers/photo/photo-renderer.component';
export * from './lib/renderers/progress/progress-renderer.component';
export * from './lib/renderers/time/time-renderer.component';
export * from './lib/renderers/timer/timer-renderer.component';
export * from './lib/renderers/todo/todo-renderer.component';
export * from './lib/renderers/user-full-name/user-full-name-renderer.component';
export * from './lib/renderers/validation-progress/validation-progress-renderer.component';
export * from './lib/renderers/video/video-renderer.component';