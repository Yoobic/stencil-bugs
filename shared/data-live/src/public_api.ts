
export { DataLiveModule } from './lib/data-live.module';

export { Channel as ChannelInterface } from './lib/interfaces/channel/channel.interface';
export * from './lib/interfaces/channels/channels.interface';
export * from './lib/interfaces/message/message.interface';
export * from './lib/interfaces/message/message.model';

export * from './lib/services/channel/channel.service';
export * from './lib/services/intercom/intercom.service';
export * from './lib/services/messages/messages.service';
export * from './lib/services/pubnub/pubnub.service';
export * from './lib/services/track/track.service';
export * from './lib/services/twilio-token/twilio-token.service';
export * from './lib/services/raven-error-handler/raven-error-handler.service';