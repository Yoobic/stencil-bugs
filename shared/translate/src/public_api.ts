export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';