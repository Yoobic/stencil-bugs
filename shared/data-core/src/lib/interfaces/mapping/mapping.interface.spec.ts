import { IMapping } from './mapping.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: Mapping', () => {
    it('should create an entity', () => {
        let entity = new IMapping();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('IMapping');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
});