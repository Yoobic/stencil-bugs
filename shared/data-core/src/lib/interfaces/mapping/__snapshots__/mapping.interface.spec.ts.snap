// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: Mapping should create an entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "autoselect": true,
      "condition": "context!=\\"translation\\"",
      "conditionalValues": Array [
        Object {
          "ADMIN_FILES_TYPE": Array [
            "locations",
            "user",
            "geofilters",
            "campaignfilters",
            "products",
          ],
          "condition": "context == \\"admin\\"",
        },
        Object {
          "condition": "context == \\"formCreator\\"",
          "values": Array [
            "pages",
          ],
        },
      ],
      "defaultValues": Array [
        "locations",
        "user",
        "geofilters",
        "campaignfilters",
        "products",
      ],
      "filterName": "type",
      "name": "type",
      "required": true,
      "title": "TYPE",
      "translate": true,
      "type": "autocomplete",
    },
    Object {
      "extensions": Array [
        "csv",
        "application",
        "xls",
        "xlsx",
      ],
      "filterName": "document",
      "filterable": false,
      "name": "document",
      "required": true,
      "title": "DOCUMENT",
      "type": "documentuploader",
    },
  ],
  "appareance": Map {},
  "className": "IMapping",
  "collectionName": undefined,
  "feathersService": undefined,
  "fields": undefined,
  "include": undefined,
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [],
  "type": [Function],
}
`;
